src/java/org/apache/cassandra/cql3/statements/AlterRoleStatement.java

```diff
@@ -114,6 +124,19 @@ public void authorize(ClientState state) throws UnauthorizedException
 
     public ResultMessage execute(ClientState state) throws RequestValidationException, RequestExecutionException
     {
+        if (opts.isGeneratedPassword())
+        {
+            String generatedPassword = Guardrails.password.generate();
+            if (generatedPassword != null)
+                opts.setOption(IRoleManager.Option.PASSWORD, generatedPassword);
+            else
+                throw new InvalidRequestException("You have to enable password_validator and it's generator_class_name property " +
```

Comment:
should we defer validation to runtime? can we do it validation during startup as well?

Reply:
@frankgh No, we should't do it here. Because imagine that you have `cassandra.yaml` "as is", that will not enable password validation. It is enabled when you explicitly enable it in `cassandra.yaml`. But we can not "dynamically enable" `GENERATED PASSWORD` stuff, that will be always there. So we need to treat the situation when a user wants to generate passwords but it is not enabled. What would we detect during startup? We do not know in advance if a user wants to use `GENERATED PASSWORD` or not. If they never use it, then there is not any problem, no?